### PR TITLE
Include console errors and warnings and get off the fork of nightmare

### PR DESCRIPTION
**Pull request by gmishkin**
This change includes a handler to report console errors and warnings other than ones coming from unhandled Javascript errors.

It also uses a different method to report the current URL to the stream so that it is not dependent on the URL coming in the event from nightmare.

If you like this method, we can add a parameter for minimum console level to report. Or, we could change it to report everything, and let the stream reader determine which severity levels to do something with.

### CODE REVIEW

File: lib/hounds.js [hunk 1]
```diff
@@ -29,6 +30,15 @@ module.exports = (options) => {
                 stackTrace
             })
         })
+        .on('console', (type, message) => {
+            if (type === 'error' || type === 'warn') {
```
Comment (justinjmoses) [inline lib/hounds.js]: I think we should just focus on `error` not `warn`

**Comment by justinjmoses** [comment]
I like the parameter to control whether or not to include `error` and `warn`

**Comment by justinjmoses** [comment]
Could you also please update the `README` so it links to the correct nightmare `3.0.1` docs

**Comment by gmishkin** [comment]
RFAL

File: README.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ To show the Electron UI, with devTools open, and keep both alive, use the follow
 * `urlFilter` An optional predicate function, taking the current `url` as a parameter, and returning `true` or `false` as to whether or not to include it in the hunt. Second argument of `domainFiltered` is a bool stating whether or not the host matches (use it if you'd like to include that check in your filter)
 * `before` and `after` callbacks receive nightmare instance and if defined, must return it (see [examples/preAuth.js](https://github.com/justinjmoses/hounds/blob/master/examples/preAuth.js#L14-L26))
 * `screnshot` A function that given the current `url`, returns the path string of the PNG to save
-* `nightmare` All [nightmare 2.10.0 options](https://github.com/segmentio/nightmare/tree/2.10.0#nightmareoptions) are supported
+* `nightmare` All [nightmare 2.10.0 options](https://github.com/segmentio/nightmare/tree/3.0.1#nightmareoptions) are supported
```
Comment (justinjmoses) [inline README.md]: can you change the link text also? :P

File: lib/hounds.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ const urlHelper = require('url')
 const Readable = require('stream').Readable
 const Nightmare = require('nightmare')
 
+const consoleLevels = new Map([
```
Comment (justinjmoses) [inline lib/hounds.js]: why not just use a regular object?

Comment (gmishkin) [reply lib/hounds.js]: Just a personal preference but basically for the reasons described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) (see section "Objects and maps compared").

Comment (justinjmoses) [reply lib/hounds.js]: It just limited the node version to 6+ (but according to this it's been there since 4.x http://node.green/).

Comment (gmishkin) [reply lib/hounds.js]: Basically use maps when the values are all the same type or objects that have the same properties.

Comment (justinjmoses) [reply lib/hounds.js]: OK - I think given where node is at now and ES6 features we should just go ahead. (I understand the general motivation - I was just being cautious about using new features without much gain)

Comment (gmishkin) [reply lib/hounds.js]: Yeah it's ES6 which is like 2015 :)

File: lib/hounds.js [hunk 4]
```diff
@@ -15,12 +21,14 @@ module.exports = (options) => {
     })
 
     let maxFollows = typeof options.maxFollows === 'number' ? options.maxFollows : Infinity
+    const consoleLevel = typeof options.consoleLevel === 'string' ? consoleLevels.get(options.consoleLevel) : 0
```
Comment (justinjmoses) [inline lib/hounds.js]: can you add this `option` to the README as well?

File: README.md [hunk 5]
```diff
@@ -65,7 +65,7 @@ To show the Electron UI, with devTools open, and keep both alive, use the follow
 * `urlFilter` An optional predicate function, taking the current `url` as a parameter, and returning `true` or `false` as to whether or not to include it in the hunt. Second argument of `domainFiltered` is a bool stating whether or not the host matches (use it if you'd like to include that check in your filter)
 * `before` and `after` callbacks receive nightmare instance and if defined, must return it (see [examples/preAuth.js](https://github.com/justinjmoses/hounds/blob/master/examples/preAuth.js#L14-L26))
 * `screnshot` A function that given the current `url`, returns the path string of the PNG to save
-* `nightmare` All [nightmare 2.10.0 options](https://github.com/segmentio/nightmare/tree/2.10.0#nightmareoptions) are supported
+* `nightmare` All [nightmare 3.0.1 options](https://github.com/segmentio/nightmare/tree/3.0.1#nightmareoptions) are supported
```
Comment (justinjmoses) [inline README.md]: are you going to a piece on the new `consoleLevel` option?

Comment (gmishkin) [reply README.md]: Yeah just been working on other stuff today.

**Comment by justinjmoses** [comment]
I tried cloning your fork, `npm i && node examples/hackerNews.js` and I got:

```
$ node examples/hackerNews.js
***** Trying url: http://news.ycombinator.com ******
***** Trying url: https://news.mit.edu/2018/manufacturing-graphene-rolls-ultrathin-membranes-0418 ******
***** Trying url: https://www.anandtech.com/show/12625/amd-second-generation-ryzen-7-2700x-2700-ryzen-5-2600x-2600 ******
/Users/jmoses/tmp/hounds/lib/hounds.js:33
            const stackTrace = stack.split('\n').map(l => l.trim())
                                     ^

TypeError: Cannot read property 'split' of undefined
    at EventEmitter.nightmare.on (/Users/jmoses/tmp/hounds/lib/hounds.js:33:38)
    at emitTwo (events.js:131:20)
    at EventEmitter.emit (events.js:214:7)
    at ChildProcess.<anonymous> (/Users/jmoses/tmp/hounds/node_modules/nightmare/lib/ipc.js:49:10)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at emit (internal/child_process.js:772:12)
    at _combinedTickCallback (internal/process/next_tick.js:141:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by justinjmoses** [comment]
I didn't have issues on my version, but then again, I didn't get any of the errors reported that I should have

**Comment by justinjmoses** [comment]
Let me know if this is RFAL - it's near merging now methinks

**Comment by gmishkin** [comment]
Yeah, agreed. It will be easier to test this out on Evergreen if it gets merged and a new release published to NPM.

**Pull request merged**